test/Exporter/Test/Writer/KnpSnappyWriterTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+// Use same namespace as system under test to mock native functions
+
+namespace Exporter\Writer;
+
+use org\bovigo\vfs\vfsStream;
+use org\bovigo\vfs\vfsStreamDirectory;
+use org\bovigo\vfs\vfsStreamWrapper;
+use Prophecy\Argument;
+
+function is_executable()
```

Comment:
> Impossible to find another way?

Without touching the code? Not sure, maybe the virtual system can be used there too…

So why is it so ugly? Well probably because we should use an abstraction to do that instead of doing ourselves. I don't think symfony/filesystem does this, do you know any lib that does?